# SPDX-FileCopyrightText: Huawei Inc.
#
# SPDX-License-Identifier: Apache-2.0

Patch for //base/hiviewdfx/faultloggerd of OpenHarmony 3.0 codebase

Configure faultloggerd service to place Unix socket in /run instead of /dev (where
they obviously does not belong).

Having a dedicated directory for this makes it easier to handle permissions in a
safe way.

Signed-off-by: Esben Haabendal <esben@geanix.com>
Upstream-Status: Inappropriate [configuration]

diff --git a/interfaces/innerkits/faultloggerd_client/faultloggerd_client.cpp b/interfaces/innerkits/faultloggerd_client/faultloggerd_client.cpp
index 704ae9c0cbd7..4456eb29b87c 100644
--- a/interfaces/innerkits/faultloggerd_client/faultloggerd_client.cpp
+++ b/interfaces/innerkits/faultloggerd_client/faultloggerd_client.cpp
@@ -31,11 +31,7 @@
 
 namespace {
 static const int32_t SOCKET_BUFFER_SIZE = 256;
-#if defined(USE_MUSL)
-static const char FAULTLOGGERD_SOCK_PATH[] = "/dev/unix/socket/faultloggerd.server";
-#else
-static const char FAULTLOGGERD_SOCK_PATH[] = "/dev/socket/faultloggerd.server";
-#endif
+static const char FAULTLOGGERD_SOCK_PATH[] = "/run/openharmony/faultlogger/server";
 }
 
 static int ReadFileDescriptorFromSocket(int socket)
